graphql-ballerina/listener_utils.bal

```diff
@@ -65,7 +68,10 @@ isolated function processJsonPayload(Engine engine, json payload, http:Response
         operationName = resolveOperationName(operationNameInPayload);
     }
     if (documentString is string) {
-        OutputObject outputObject = engine.getOutputObjectForQuery(documentString, operationName);
+        [OutputObject, string] [outputObject, errorType] = engine.getOutputObjectForQuery(documentString, operationName);
+        if (errorType == "SyntaxError" || errorType == "DocumentError") {
+            response.statusCode = 400;
```

Comment:
Instead of setting the status code directly, I guess we can directly use the `http:BAD_REQUEST` response here, but I'm not sure. Can you check that as well?